src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/_params.py

```diff
@@ -0,0 +1,139 @@
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core.commands import \
+    (register_cli_argument, register_extra_cli_argument, CliArgumentType)
+import azure.cli.core.commands.arm  # pylint: disable=unused-import
+from azure.cli.core.commands.validators import get_default_location_from_resource_group
+from azure.cli.core.commands.parameters import (
+    resource_group_name_type,
+    location_type,
+    tags_type,
+    deployment_name_type,
+    enum_choice_list,
+    get_resource_name_completion_list
+)
+from azure.cli.core._profile import Profile
+from azure.cli.core.util import get_json_object
+
+from azure.cli.core.sdk.util import ParametersContext
+
+# PARAMETER REGISTRATIONS
+
+register_cli_argument('sf cluster list', 'resource_group_name', resource_group_name_type,
+                      id_part=None, required=False, help='The resouce group name')
+
+register_cli_argument('sf', 'resource_group_name', resource_group_name_type,
+                      id_part=None, required=True, help='The resouce group name')
+register_cli_argument('sf', 'cluster_name',options_list=('--name', '--cluster-name','-n'),
+                      help='Specify the name of the cluster, if not given it will be same as resource group name')
+register_cli_argument('sf', 'location',
+                      validator=get_default_location_from_resource_group)
+
+register_cli_argument('sf', 'secret_identifier',options_list=('--secret-identifier', '-si'),
+                      help='The existing Azure key vault secret URL')
+register_cli_argument('sf', 'certificate_file',options_list=('--certificate-file', '-cf'),
+                     help='The existing certificate file path for the primary cluster certificate.')
+register_cli_argument('sf', 'parameter_file',options_list=('--parameter-file', '-pf'),
+                     help='The path to the template parameter file.')
+register_cli_argument('sf', 'template_file',options_list=('--template-file', '-tf'),
+                     help='The path to the template file.')
+register_cli_argument('sf', 'vm_password',options_list=('--vm-password', '-vp'),
+                      help='The password of the Vm')
+register_cli_argument('sf', 'certificate_output_folder',options_list=('--certificate-output-folder', '-cof'),
+                      help='The folder of the new certificate file to be created.')
+register_cli_argument('sf', 'certificate_password',options_list=('--certificate-password', '-cp'),
+                      help='The password of the certificate file.')
+register_cli_argument('sf', 'certificate_subject_name',options_list=('--certificate-subject-name', '-sn','-csn'),
+                      help='The subject name of the certificate to be created.')
+register_cli_argument('sf', 'vault_resource_group_name',options_list=('--vault-resource-group', '-vg','-vrg'),
```

Comment:
Yes, short options must only be a single character. If they are multi-character, they should be preceded by `--`.  And it is highly unusually to have multiple long-options in the CLI.  Only when one is going to be deprecated in a future release.